### PR TITLE
Initial support of rootfs on ceph rbd

### PR DESCRIPTION
**Pull request by bitphage**
I did not created a tests yet because I had used separate testing environment deployed via vagrant + ansible including ceph cluster. If tests are mandatory I'll postpone PR.

### CODE REVIEW

File: modules.d/95rbd/rbdroot.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    rbd_bus=/sys/bus/rbd/add
+else
+    echo "ERROR: /sys/bus/rbd/add does not exist"
+    return 1
+fi
+
+# tell the kernel rbd client to map the block device
+echo "$mons name=$user,secret=$key $pool $image $snap" > $rbd_bus
+# figure out where the block device appeared
+dev=$(ls /dev/rbd* | grep '/dev/rbd[0-9]*$' | tail -n 1)
```
Comment (haraldh) [inline modules.d/95rbd/rbdroot.sh]: With:
```
dev=$(ls -1art /dev/rbd[0-9]* | (while read l; do dev=$l ;done; echo $dev))
```
```grep``` and ```tail``` are not needed.

File: modules.d/95rbd/rbdroot.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+# add partition if set
+if [ $partition ]; then
+    dev=${dev}p$partition
+fi
+
+mount -t $fstype $dev $NEWROOT -o $opts
```
Comment (haraldh) [inline modules.d/95rbd/rbdroot.sh]: Manual mounting should only be necessary, if ```root=rbd:...```. In the case of ```netroot=rbd:...```, which should be the preferred way, ```root=UUID=...``` or ```root=LABEL=...``` should let dracut or systemd mount the disk just fine on ```$NEWROOT```.

File: modules.d/95rbd/module-setup.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+# called by dracut
+installkernel() {
+    instmods rbd aes cbc
+}
+
+# called by dracut
+install() {
+    inst_multiple grep tail
```
Comment (haraldh) [inline modules.d/95rbd/module-setup.sh]: Might not be necessary, as stated below.

**Comment by haraldh** [comment]
ping?

**Comment by bitfag** [comment]
Sorry it was a one-time project for me, maybe someone else will do required changes & necessary testing.

**Pull request closed without merging**